lib/Dialect/FIRRTL/Transforms/CreateSiFiveMetadata.cpp

```diff
@@ -32,6 +32,19 @@ static const char metadataDirectoryAnnoClass[] =
     "sifive.enterprise.firrtl.MetadataDirAnnotation";
 
 namespace {
+
+// TODO: Move this to firrtl/utility
+StringAttr getOrAddInnerSym(Operation *op) {
+  auto attr = op->getAttrOfType<StringAttr>("inner_sym");
+  if (attr)
+    return attr;
+  ModuleNamespace modNamespace(op->getParentOfType<FModuleOp>());
+  auto name = modNamespace.newName("s_");
```

Comment:
Reconstructing the module namespace every time might be slow.  We have this common block of code that we have been copy-pasting between passes that can help:

https://github.com/llvm/circt/blob/e4ecc625a06af655555f420c2c41f779650115c6/lib/Dialect/FIRRTL/Transforms/AddSeqMemPorts.cpp#L42-L68